bin/clang/astConsumers.cc

```diff
@@ -65,24 +65,13 @@ SkeletonASTConsumer::HandleTopLevelDecl(DeclGroupRef DR)
 {
   for (DeclGroupRef::iterator b = DR.begin(), e = DR.end(); b != e; ++b){
     Decl* d = *b;
-    switch(d->getKind()){
-     case Decl::Function: {
-        FunctionDecl* fd = cast<FunctionDecl>(d);
-        //the function decl will have its body filled in later
-        //possibly - we need to make sure to only add the function once
-        if (fd->isThisDeclarationADefinition()){
-          //also, we only really care about the definition anyway
-          allTopLevelDecls_.push_back(d);
-        }
-        if (isNullWhitelisted(fd->getNameAsString())){
-          CompilerGlobals::astNodeMetadata.nullSafeFunctions[fd] = nullptr;
-        }
+    if (d->getKind() == Decl::Function){
```

Comment:
The typical idiom for this kind of thing is
```cpp 
if(FunctionDecl const* fd = dyn_cast<FunctionDecl>(d)){
}
```

Reply:
If you check out the source of `dyn_cast` it is basically just a type check followed by a cast. I wouldn't port this assignment in the `if` to something like Kokkos, but it's idiomatic in LLVM.